lib/resty/core/base.lua

```diff
@@ -69,14 +74,38 @@ end
 do
     local orig_require = require
     local pkg_loaded = package.loaded
-    local function my_require(name)
-        local mod = pkg_loaded[name]
-        if mod then
-            return mod
+    local key_sentinel
+
+    local function get_key_sentinel()
+        local n = C.lj_ffi_get_key_sentinel()
+        key_sentinel = tonumber(n)
```

Comment:
Why the `tonumber`? In older versions of LuaJIT sentinel was a userdata value. This could be change again.